test/data/BinaryHeap.ts

```diff
@@ -0,0 +1,740 @@
+          checkNodes(await instance['toArray()']());
+        });
+
+        it('returns true if value is added', async () => {
+          expect(await instance.callStatic['add(bytes32)'](zeroBytes32)).to.be
+            .true;
+        });
+
+        it('returns false if value has already been added', async () => {
+          await instance.callStatic['remove(bytes32)'](zeroBytes32);
+          expect(await instance.callStatic['remove(bytes32)'](zeroBytes32)).to
+            .be.false;
+        });
+      });
+
+      describe('#remove(bytes32)', () => {
+        it('sets the parent node such that it is greater than or equal to the values of its children when a node is removed', async () => {
+          await instance['add(bytes32)'](twoBytes32);
+          await instance['add(bytes32)'](zeroBytes32);
+          await instance['add(bytes32)'](eightBytes32);
+          await instance['add(bytes32)'](threeBytes32);
+          await instance['add(bytes32)'](sixBytes32);
+          await instance['add(bytes32)'](fourBytes32);
+          await instance['add(bytes32)'](fiveBytes32);
+          await instance['add(bytes32)'](nineBytes32);
+          await instance['add(bytes32)'](sevenBytes32);
+          await instance['add(bytes32)'](oneBytes32);
+
+          await instance['add(bytes32)'](sixBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](fiveBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(bytes32)'](eightBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](twoBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](fourBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](zeroBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](threeBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](nineBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['remove(bytes32)'](oneBytes32);
+          checkNodes(await instance['toArray()']());
+
+          await instance['add(bytes32)'](sevenBytes32);
+          checkNodes(await instance['toArray()']());
+        });
+
+        it('returns true if value is removed', async () => {
+          await instance['add(bytes32)'](zeroBytes32);
+          await instance['add(bytes32)'](oneBytes32);
+          await instance['add(bytes32)'](twoBytes32);
+
+          expect(await instance.callStatic['remove(bytes32)'](zeroBytes32)).to
+            .be.true;
+        });
+
+        it('returns false if value does not exist', async () => {
+          expect(await instance.callStatic['remove(bytes32)'](zeroBytes32)).to
+            .be.false;
+        });
```

Comment:
Up to you whether to add a test which checks that the highest value is the root, when an element is removed.

I guess this is included in `checkNodes` although explicit root checking may be worthwhile.

Reply:
added in [76247ee6](https://github.com/solidstate-network/solidstate-solidity/pull/134/commits/76247ee61179a71413f271013c39573ef116b430)